extensions/opentelemetry/runtime/src/main/java/io/quarkus/opentelemetry/runtime/propagation/OpenTelemetryContextProvider.java

```diff
@@ -0,0 +1,38 @@
+package io.quarkus.opentelemetry.runtime.propagation;
+
+import java.util.Map;
+
+import org.eclipse.microprofile.context.spi.ThreadContextProvider;
+import org.eclipse.microprofile.context.spi.ThreadContextSnapshot;
+
+import io.quarkus.opentelemetry.runtime.QuarkusContextStorage;
+
+public class OpenTelemetryContextProvider implements ThreadContextProvider {
```

Comment:
Please rename this to OpenTelemetryMpContextPropagationProvider, or similar.
OTel itself has a context and this can be misleading

Reply:
@brunobat It is now renamed. Let me know if it fits your expectations.

Reply:
It's good. Roberto's comment is important. Will wait for that before approving.